src/server_manager/model/server.ts

```diff
@@ -37,6 +37,15 @@ export interface Server {
   // Removes the access key given by id.
   removeAccessKey(accessKeyId: AccessKeyId): Promise<void>;
 
+  // Sets a data transfer limit over a 30 day rolling window for all access keys.
+  setAccessKeyDataLimit(limit: DataLimit): Promise<void>;
+
+  // Returns the access key data transfer limit, or undefined if it has not been set.
+  getAccessKeyDataLimit(): DataLimit|undefined;
```

Comment:
We should return `|null` instead of `|undefined`, it's more explicit.

We should avoid `undefined` whenever possible. `undefined` is usually an unassigned variable, a return without a value or a missing field in a structure. Things that a usual compiler would complain about. It's an unintentional value.

`null` is an explicit intention to represent the absence of something. You have the guarantee that someone wrote an assignment or a return statement with null. It's well defined.

For those reasons, I believe it's better to use null. However.... I see that we are using `undefined` in other functions, so we may want to keep for consistency, or change all of them. Up to you.

Reply:
I had similar thoughts when I first looked at this code but `undefined` does somewhat make sense if the server isn't using data limits.

My actual preference would be to explicitly use an Optional<T> type, something like`type Optional<T> = T|undefined` which would play well with [TS 3.7's null coalescing and optional chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining), and which clearly states the semantics of the returned thing.

Reply:
I have no strong opinions about this, but I do think `undefined` is semantically correct in this case; the server has not defined a value for the data limit and I can see a scenario where null takes a different meaning.